WordPress/Sniffs/WP/DeprecatedParameterValuesSniff.php

```diff
@@ -0,0 +1,196 @@
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $target_functions = array(
+	 *         (string) Function name. => array(
+	 *             (int) Target parameter position, 1-based. => array(
+	 *                 (string) Parameter value. => array(
+	 *                     'alt'     => (string) Suggested alternative.
+	 *                     'version' => (int) The WordPress version when deprecated.
+	 *                 )
+	 *             )
+	 *         )
+	 *     );
+	 */
+	protected $target_functions = array(
+		'add_settings_field' => array(
+			4 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+		'add_settings_section' => array(
+			4 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+		'bloginfo' => array(
+			1 => array(
+				'home' => array(
+					'alt'     => 'echo esc_url( home_url() )',
+					'version' => '2.2.0',
+				),
+				'siteurl' => array(
+					'alt'     => 'echo esc_url( home_url() )',
+					'version' => '2.2.0',
+				),
+				'text_direction' => array(
+					'alt'     => 'is_rtl()',
+					'version' => '2.2.0',
+				),
+			),
+		),
+		'get_bloginfo' => array(
+			1 => array(
+				'home' => array(
+					'alt'     => 'home_url()',
+					'version' => '2.2.0',
+				),
+				'siteurl' => array(
+					'alt'     => 'home_url()',
+					'version' => '2.2.0',
+				),
+				'text_direction' => array(
+					'alt'     => 'is_rtl()',
+					'version' => '2.2.0',
+				),
+			),
+		),
+		'register_setting' => array(
+			1 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+		'unregister_setting' => array(
+			1 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		$this->get_wp_version_from_cl();
+		$param_count = count( $parameters );
```

Comment:
I've just implemented using FQN for functions which can benefit from it in #1414. If/when you are updating the PR anyway, it would be appreciated if you could change this function call to `\count()` (just add the backslash).